integration_tests/src/main/python/map_test.py

```diff
@@ -32,6 +38,44 @@ def test_simple_get_map_value(data_gen):
                 'a["NOT_FOUND"]',
                 'a["key_5"]'))
 
```

Comment:
I think a very common case is going to be scalar keys (especially strings). Could we please test this case too?

Reply:
I spent some time looking into this and we do not support this yet. I filed https://github.com/NVIDIA/spark-rapids/issues/3245 to implement it.

Reply:
Sorry about the confusion. I think I put the comment in a bad place.  What I think is going to be common is something like.

```
map("key1", valueColumn)
```

This is what I want a test for. From the code it looks like it should just work all the time, but it would be nice to have an explicit test for this.

Reply:
That makes more sense. I added the test and it did expose an issue, which is now fixed.